### PR TITLE
LW-34 Messages in nonexisting loggers are handled improperly

### PR DESCRIPTION
**Pull request by volhovm**

### CODE REVIEW

**Comment by avnik** [comment]
LGTM :+1:

File: src/System/Wlog/IOLogger.hs [hunk 1]
```diff
@@ -117,21 +118,24 @@ logInternalState = unsafePerformIO $ do
...
+Given a name, return all components of it, starting from the root.
+
 Example return value:
 
->["", "MissingH", "System.Cmd.Utils", "System.Cmd.Utils.pOpen"]
+λ> componentsOfName (LoggerName "a.b.c")
+[LoggerName {getLoggerName = ""},
+ LoggerName {getLoggerName = "a"},
+ LoggerName {getLoggerName = "a.b"},
+ LoggerName {getLoggerName = "a.b.c"}]
```
Comment (chshersh) [inline src/System/Wlog/IOLogger.hs:130]: Really nice example! I like to see documentation improvements :+1:

File: src/System/Wlog/IOLogger.hs [hunk 2]
```diff
@@ -156,63 +160,67 @@ logMCond logname sev msg cond = do
...
+    parentLoggers :: LoggerName -> m [Logger]
+    parentLoggers = fmap reverse . mapM getLogger . componentsOfName
+
+    -- Tries to log the message into handlers. sevFiltPassed variable
+    -- denotes the "has log message passed through the first severity
+    -- filter". We only apply severity filter once, the first time we
+    -- encounter it.
+    traverseAndLog :: Bool -> [Logger] -> m ()
+    traverseAndLog _ [] = pass
+    traverseAndLog sevFiltPassed (x:xs) = do
```
Comment (chshersh) [inline src/System/Wlog/IOLogger.hs]: [`whenNotNull`](https://github.com/serokell/universum/blob/eec7d0dddd8ead098f6b177c2222a3d2475b9119/src/Universum/List/Safe.hs#L64-L66) from `universum` can be used here

File: src/System/Wlog/IOLogger.hs [hunk 3]
```diff
@@ -156,63 +160,67 @@ logMCond logname sev msg cond = do
...
+        let doLog n = do
+                forM_ (x ^. lHandlers) callHandler
+                traverseAndLog n xs
+        case x ^. lLevel of
+           _         | sevFiltPassed -> doLog sevFiltPassed
```
Comment (chshersh) [inline src/System/Wlog/IOLogger.hs]: I want to say that this `case-of` expression is written in obscure way.

File: src/System/Wlog/IOLogger.hs [hunk 4]
```diff
@@ -156,63 +160,67 @@ logMCond logname sev msg cond = do
...
+           Nothing   -> doLog sevFiltPassed
+           (Just lp) | sev `Set.member` lp -> doLog True
```
Comment (chshersh) [inline src/System/Wlog/IOLogger.hs]: For example, here you could use `when` function instead of guards.

File: src/System/Wlog/IOLogger.hs (hunk 3 above)
Comment (volhovm) [reply src/System/Wlog/IOLogger.hs]: ~fixed

File: src/System/Wlog/IOLogger.hs (hunk 2 above)
Comment (volhovm) [reply src/System/Wlog/IOLogger.hs]: Added.

**Comment by volhovm** [comment]
@ChShersh CI doesn't pass, but it also doesn't in master (the failure is not related to this change), so can we merge it?

**Pull request merged**